### PR TITLE
Feature/product detail page

### PR DESCRIPTION
**Pull request by KazeUmiAI**
# Description

Describe what you have done in this PR.

# Checklist

- [ ] This is a breaking changes?

# Screenshots

Provide screenshots or videos of the changes made if any.

Please provide screenshots or videos:

- From `playground` if you making changes in `@namviet-fe/core-ui`
- From `next js app` if your making any changes in `@namviet-fe/next-app`.

### CODE REVIEW

File: packages/core-components/src/components/product-detail/ProductCarousel.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                  alt={item.alt}
+                />
+              </span>
+            );
+          },
+        };
+      });
+
+    return (
+      <div className=''>
```
Comment (its-rav) [inline packages/core-components/src/components/product-detail/ProductCarousel.tsx]: redundant

File: packages/core-components/src/components/product-detail/ProductCarousel.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          renderLeftNav={(onClick, disabled) => (
+            <button
+              onClick={onClick}
+              disabled={disabled}
+              type='button'
+              className='image-gallery-icon image-gallery-left-nav'
+              aria-label='Previous Slide'
+              style={{ padding: '10px', backgroundColor: '#4e4e4e' }}
+            >
+              <svg
```
Comment (its-rav) [inline packages/core-components/src/components/product-detail/ProductCarousel.tsx:75]: can we move this to seperate file/component for better reading ?

File: packages/core-components/src/components/product-detail/ProductSpecifications.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        <div className='absolute bottom-0 left-0 right-0 h-[40px] bg-gradient-to-t from-white'></div>
+      </div>
+      <div className='flex justify-center py-[40px] hover:text-blue-600'>
+        <button
+          className='flex items-center gap-[5px]'
+          onClick={() => {
+            setIsShowPopUp(true);
+          }}
+        >
+          <span className='font-medium'>{viewMoreLabel}</span>
```
Comment (its-rav) [inline packages/core-components/src/components/product-detail/ProductSpecifications.tsx:132]: we can do static translation instead of this

Comment (KazeUmiAI) [reply packages/core-components/src/components/product-detail/ProductSpecifications.tsx:132]: use this prop for static translation will be handle at page, not in component

File: packages/core-components/src/components/product-detail/ProductCarousel.tsx (hunk 2 above)
Comment (KazeUmiAI) [reply packages/core-components/src/components/product-detail/ProductCarousel.tsx:75]: We can but didn't neccesarily

File: packages/core-components/src/components/product-detail/ProductSpecifications.tsx (hunk 3 above)
Comment (its-rav) [reply packages/core-components/src/components/product-detail/ProductSpecifications.tsx:132]: pls use what we already have a.

File: packages/core-components/package.json [hunk 4]
```diff
@@ -31,6 +32,9 @@
     "@types/react": "^18.0.37",
     "@types/react-dom": "^18.0.11",
     "@types/react-google-recaptcha": "^2.1.8",
+    "@types/react-image-gallery": "^1.2.4",
+    "@typescript-eslint/eslint-plugin": "^5.59.0",
```
Comment (its-rav) [inline packages/core-components/package.json:37]: why we need this packages a

File: packages/core-components/src/components/contact/ContactMap.tsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ type ContactMapProps = {
 
 const ContactMap: React.FC<ContactMapProps> = ({ contactMapItem }) => {
   return contactMapItem?.srcMap ? (
-    <div className=''>
+    <div>
```
Comment (its-rav) [inline packages/core-components/src/components/contact/ContactMap.tsx:13]: 💯

File: packages/core-components/src/components/product-detail/ProductCarousel.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+};
+
+type ProductCarouselProps = {
+  productImgItems: ProductImgItemType[];
+};
+
+const ProductCarousel: React.FC<ProductCarouselProps> = ({
+  productImgItems,
+}) => {
+  const slideShowRef = useRef<any>();
```
Comment (its-rav) [inline packages/core-components/src/components/product-detail/ProductCarousel.tsx:18]: why we need assert `any` here a

File: packages/core-components/src/components/product-detail/ProductCarousel.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+                  alt={item.alt}
+                />
+              </span>
+            );
+          },
+        };
+      });
+
+    return (
+      <div>
```
Comment (its-rav) [inline packages/core-components/src/components/product-detail/ProductCarousel.tsx:50]: redundant div

File: packages/core-components/src/components/product-detail/ProductCarousel.tsx [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+          }}
+          onScreenChange={() => {
+            !isFullScreen ? setIsFullScreen(true) : setIsFullScreen(false);
+          }}
+          renderLeftNav={(onClick, disabled) => (
+            <button
+              onClick={onClick}
+              disabled={disabled}
+              type='button'
+              className='image-gallery-icon image-gallery-left-nav'
```
Comment (its-rav) [inline packages/core-components/src/components/product-detail/ProductCarousel.tsx:71]: why we need all of this random class, a? is it required from the package?

File: packages/core-components/src/components/product-detail/ProductCarousel.tsx (hunk 7 above)
Comment (KazeUmiAI) [reply packages/core-components/src/components/product-detail/ProductCarousel.tsx:50]: removed

**Review by its-rav — APPROVED** [review]

**Review by its-rav — APPROVED** [review]

**Pull request merged**